go/private/rules/binary.bzl

```diff
@@ -213,6 +213,7 @@ _go_binary_kwargs = {
             doc = """List of Go libraries this package imports directly.
             These may be `go_library` rules or compatible rules with the [GoLibrary] provider.
             """,
+            cfg = go_transition,
```

Comment:
You should be able to remove the `non_go_transition` configurations: Since the binary rule itself isn't transitioned anymore, there is no need to reset Go settings.

Same for `go_test` below.